src/ssort/_requirements.py

```diff
@@ -191,3 +210,21 @@ def _get_requirements_for_name(node: ast.Name) -> Iterable[Requirement]:
         yield Requirement(
             name=node.id, lineno=node.lineno, col_offset=node.col_offset
         )
+
+
+if sys.version_info >= (3, 12):
+
+    @get_requirements.register(ast.TypeAlias)
+    def _get_requirements_for_type_alias(
+        node: ast.TypeAlias,
+    ) -> Iterable[Requirement]:
+        for type_param in node.type_params:
+            yield from get_requirements(type_param)
+        scope: set[str] = set()
+        scope.update(type_param.name for type_param in node.type_params)  # type: ignore[attr-defined]
+        scope.add(node.name.id)
+        for requirement in get_requirements(node.value):
+            if not requirement.deferred:
+                requirement = dataclasses.replace(requirement, deferred=True)
```

Comment:
I've copied the two lines above from the similar code in `_get_requirements_for_function_def`. I don't know what a referred requirement is or if this code is needed here.

Reply:
This is good.  Might be worth factoring out a `_get_scope_from_type_params` function.

Reply:
A deferred requirement is one that is captured by a statement but not used immediately.

For example here, `a` has an immediate requirement on `b`:
```python
a = b
```
and here the requirement is deferred:
```python
def a():
    return b
```